src/Tools/dotnet-dsrouter/Program.cs

```diff
@@ -104,6 +112,58 @@ internal sealed class Program
                 IpcClientAddressOption(), TcpClientAddressOption(), RuntimeTimeoutOption(), VerboseOption(), ForwardPortOption()
             };
 
+        private static Command IOSSimulatorRouterCommand() =>
+            new(
+                name: "ios-sim",
+                description: "Start a .NET application Diagnostics Server routing local IPC server <--> iOS Simulator. " +
+                                "Router is configured using an IPC server (connecting to by diagnostic tools) " +
+                                "and a TCP/IP server (accepting runtime TCP client).")
+            {
+                // Handler
+                HandlerDescriptor.FromDelegate((DiagnosticsServerIpcServerIOSSimulatorRouterDelegate)new DiagnosticsServerRouterCommands().RunIpcServerIOSSimulatorRouter).GetCommandHandler(),
+                // Options
+                RuntimeTimeoutOption(), VerboseOption()
+            };
+
+        private static Command IOSRouterCommand() =>
+            new(
+                name: "ios",
+                description: "Start a .NET application Diagnostics Server routing local IPC server <--> iOS Device over usbmux. " +
+                                "Router is configured using an IPC server (connecting to by diagnostic tools) " +
+                                "and a TCP/IP client (connecting runtime TCP server over usbmux).")
+            {
+                        // Handler
+                        HandlerDescriptor.FromDelegate((DiagnosticsServerIpcServerIOSRouterDelegate)new DiagnosticsServerRouterCommands().RunIpcServerIOSRouter).GetCommandHandler(),
+                        // Options
+                        RuntimeTimeoutOption(), VerboseOption()
+            };
+
+        private static Command AndroidEmulatorRouterCommand() =>
+            new(
+                name: "android-emu",
```

Comment:
I've always used identical commands for Android emulator vs device. Do we need `android-emu` as an option?

Or is the idea to just support it and do the same thing?

Reply:
In the https://github.com/dotnet/runtime/blob/main/docs/design/mono/diagnostics-tracing.md I actually documented the emulator to use the 10.0.2.2 IP address so you don't need to depend on having a reverse port adb command running that you would need for a device. In order to have that option available I split android-emu and android into two different things, but you could of course use android for both scenarios, just that the tooling will depend on more things running when not really needed.

Reply:
android-emu option will log out a preconfigured DOTNET_DiagnosticPorts= that uses 10.0.2.2 instead of 127.0.0.1 that would need adb reverse ports running.

android option will log out a preconfigured DOTNET_DiagnosticPorts= that uses 127.0.0.1 and it will use the forward-port argument automatically setting up needed adb reverse ports needed for that configuration to work.

In the end, you can use android to work on both emulator and device, just that the emulator scenario could be simplified and not depend on running adb reverse port command.